sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/collectionOperations.scala

```diff
@@ -2623,8 +2623,16 @@ object Sequence {
         // about a month length in days and a day length in microseconds
         val intervalStepInMicros =
           stepMicros + stepMonths * microsPerMonth + stepDays * microsPerDay
-        val startMicros: Long = num.toLong(start) * scale
-        val stopMicros: Long = num.toLong(stop) * scale
+
+        // Date to timestamp is not equal from GMT and Chicago timezones
+        val (startMicros, stopMicros) = if (scale == 1) {
+          (num.toLong(start), num.toLong(stop))
+        }
+        else {
+          (daysToMicros(num.toInt(start), zoneId),
+            daysToMicros(num.toInt(stop), zoneId))
```

Comment:
can you explain a bit more? It's hard to understand this change without any comment.

Reply:
Yeh, please, explain why if `scale` != 1, `start` and `stop` contain days.

Reply:
Because when `scale` != 1，it is converted to day count，so we may need to use zone info to translate into microseconds to get a correct result，rather than just multiply `MICROS_PER_DAY` which ignore timezone.

Reply:
> Because when scale != 1，it is converted to day count

How can we tell it?

Reply:
maybe we should add more documents to `TemporalSequenceImpl` first, to understand what it is doing

Reply:
Done, maybe we can pass the scale through constructor：
private class TemporalSequenceImpl[T: ClassTag]
      (dt: IntegralType, **scale: Long**, fromLong: Long => T, zoneId: ZoneId)